src/compiler/types.ts

```diff
@@ -2085,8 +2086,23 @@ namespace ts {
         type: TypeNode;
     }
 
-    export interface JSDocFunctionType extends JSDocType, SignatureDeclaration {
+    export interface JSDocFunctionType extends JSDocType, Declaration {
         kind: SyntaxKind.JSDocFunctionType;
+        parameters: NodeArray<JSDocFunctionTypeParameterDeclaration>;
+        type: TypeNode;
+    }
+
+    export const enum JSDocFunctionTypeParameterSort {

```

Comment:
-Kind is a better suffix than -Sort here.

Reply:
Even better: `name: JSDocParameterSpecialName`, with Regular renamed to None.

Reply:
or specialName so that it isn't structurally compatible with other named things.